pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -61,17 +210,203 @@ func (r *Builder) VirtualMachine(vmRef ref.Ref, object *cnv.VirtualMachineSpec,
+		cpuThreadPolicy = imageCPUThreadPolicy.(string)
+	}
+	if cpuThreadPolicy == CPUThreadPolicyIsolate {
+		object.Template.Spec.Domain.CPU.IsolateEmulatorThread = true
+	}
+
+	// Set CPU Sockets/Cores/Threads and Memory requests
+	var cpuSockets, cpuCores, cpuThreads int
+
+	cpuSockets = vm.Flavor.VCPUs
+	cpuCores = 1
+	cpuThreads = 1
+
+	if flavorCPUSockets, ok := vm.Flavor.ExtraSpecs[FlavorCPUSockets]; ok {
+		if sockets, err := strconv.Atoi(flavorCPUSockets); err == nil {
+			cpuSockets = sockets
+		}
+	} else if imageCPUSockets, ok := vm.Image.Properties[CPUSockets]; ok {
+		if sockets, ok := imageCPUSockets.(int); ok {
+			cpuSockets = sockets
+		}
+	}
+	if flavorCPUCores, ok := vm.Flavor.ExtraSpecs[FlavorCPUCores]; ok {
+		if cores, err := strconv.Atoi(flavorCPUCores); err == nil {
+			cpuCores = cores
+		}
+	} else if imageCPUCores, ok := vm.Image.Properties[CPUCores]; ok {
+		if cores, ok := imageCPUCores.(int); ok {
+			cpuCores = cores
+		}
+	}
+	if flavorCPUThreads, ok := vm.Flavor.ExtraSpecs[FlavorCPUThreads]; ok {
+		if threads, err := strconv.Atoi(flavorCPUThreads); err == nil {
+			cpuThreads = threads
+		}
+	} else if imageCPUThreads, ok := vm.Image.Properties[CPUThreads]; ok {
+		if threads, ok := imageCPUThreads.(int); ok {
+			cpuThreads = threads
+		}
+	}
+
+	resourceRequests := map[core.ResourceName]resource.Quantity{}
+
+	if cpuCores != 1 || cpuThreads != 1 {
+		object.Template.Spec.Domain.CPU.Sockets = uint32(cpuSockets)
+		object.Template.Spec.Domain.CPU.Cores = uint32(cpuCores)
+		object.Template.Spec.Domain.CPU.Threads = uint32(cpuThreads)
+	} else {
+		cpuRequest := resource.NewQuantity(int64(cpuSockets*cpuCores*cpuThreads), resource.BinarySI)
+		resourceRequests[core.ResourceCPU] = *cpuRequest
+	}
+
+	memoryRequest := resource.NewQuantity(int64(vm.Flavor.RAM*int(resource.Mega)), resource.BinarySI)
+	resourceRequests[core.ResourceMemory] = *memoryRequest
+
+	object.Template.Spec.Domain.Resources.Requests = resourceRequests
+
+	// Set Max CPU Sockets/Cores/Threads
+	var maxCPUSockets, maxCPUCores, maxCPUThreads int
+
+	maxCPUSockets = cpuSockets
+	maxCPUCores = cpuCores
+	maxCPUThreads = cpuThreads
+
+	if flavorMaxCPUSockets, ok := vm.Flavor.ExtraSpecs[FlavorMaxCPUSockets]; ok {
+		if sockets, err := strconv.Atoi(flavorMaxCPUSockets); err == nil {
+			maxCPUSockets = sockets
+		}
+	}
+	if flavorMaxCPUCores, ok := vm.Flavor.ExtraSpecs[FlavorMaxCPUCores]; ok {
+		if cores, err := strconv.Atoi(flavorMaxCPUCores); err == nil {
+			maxCPUCores = cores
+		}
+	}
+	if flavorMaxCPUThreads, ok := vm.Flavor.ExtraSpecs[FlavorMaxCPUThreads]; ok {
+		if threads, err := strconv.Atoi(flavorMaxCPUThreads); err == nil {
+			maxCPUThreads = threads
+		}
+	}
+
+	cpuLimit := resource.NewQuantity(int64(maxCPUSockets*maxCPUCores*maxCPUThreads), resource.BinarySI)
+	object.Template.Spec.Domain.Resources.Limits = map[core.ResourceName]resource.Quantity{
+		core.ResourceCPU: *cpuLimit,
```

Comment:
I don't think that's the right mapping - cpu-limit in k8s specifies the maximum consumption of cpu shares the pod can get while in openstack it seems like in ovirt - the (max) topology that defines how many cpus can be hot-plugged to the vm

Reply:
I have removed this part